posts/2020/07/the-ibis-backends.md

```diff
@@ -0,0 +1,411 @@
+      <td></td>
+      <td>9</td>
+      <td>2</td>
+      <td>2</td>
+    </tr>
+    <tr>
+      <th>2019</th>
+      <td>33</td>
+      <td>22</td>
+      <td>17</td>
+      <td>12</td>
+      <td>32</td>
+      <td>1</td>
+      <td>4</td>
+      <td></td>
+      <td>7</td>
+      <td>1</td>
+      <td>2</td>
+      <td>5</td>
+    </tr>
+    <tr>
+      <th>2020</th>
+      <td>38</td>
+      <td>3</td>
+      <td>4</td>
+      <td>2</td>
+      <td>4</td>
+      <td>1</td>
+      <td>2</td>
+      <td>1</td>
+      <td>3</td>
+      <td>4</td>
+      <td>4</td>
+      <td></td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+> We note an early focus `ibis.sqlite, sqlalchemy and ibis.impala`. 
+Later, work began on the `pandas` backend rounding out the three different types of backgrounds.
+From this point, improvements were made to these key backends as `ibis.clickhouse, ibis.spark and "postgres"`. Recently, 
+For the past 3 years, Quansight, in partnership with OmniSci, added the `ibis.omniscidb`
+string generating backend. Since Quansight Labs has taken on a role as a community maintainer 
+for `ibis`. This collaboration introduced geospatial functionality to `ibis`, and we 
+have on going efforts to introduce [SQL Server][sql-server] support
+
+### Currently, there is an ongoing effort to add sqlserver backends. 
+
+Leave a comment the backends what you like to see? `dask`? `altair`?
+
+## `ibis` direct execution.
+
+`ibis` direct execution backends like `pandas and hdf5` operate on conventional in-memory python objects.
+`pandas` is the gold standard for structured data in python, and inspires the api for `ibis`.
+
+`pd` is an `ibis` backend based off `pandas.DataFrame` objects.
+
+`expression` is an `ibis` query, that has `expression.compile` and `expression.execute` methods.
+We'll recognize the __execute__ method when we return `pandas.DataFrame`s from `ibis` expression.
+The __compile__ method does not trigger any computation, rather it constructs an intermediate form
+that is interpretted by a backend.
+
+In the case of direction execution backends, the `expression` compiles to an the original `ibis` 
+expression.  And the computation is carried out based on a set of recipes defined in `ibis`.
+
+In general, we would typically do this work directly in `pandas`, however this work is
+practical in mocking tests for expressions independent of backends.
+
+> Learn more about the [HDF5 direct execution backend in the `ibis` tests][test-hdf5].
+
+## `ibis` expression generating backends.
+
+Expression generating backends operate on [SQL] databases that interoperator with `sqlalchemy`.
+
+    
+In the case of expression generating backends, the intermediate representation is a `sqlalchemy` object.
+`sqlalchemy` is _The Database Toolkit for Python_, and `ibis` leverages it compatability
+with traditional [SQL] databases.
+    
+
+
+## `ibis` string generating backends.
+
+```bash
+pip install --upgrade ibis-framework[omniscidb]
```

Comment:
So I was recently led astray by the ibis docs a few weeks ago. I didn't realize that if you install `ibis-framework` from conda-forge, you get ALL the backends. I'd also add that comment here `conda install -c conda-forge ibis-framework`